apps/ui/src/Communities/CommunityCard/StatusButton.tsx

```diff
@@ -1,18 +1,16 @@
 import { Box, IconButton, Tooltip } from '@mui/material';
-import {
-  AddRounded,
-} from '@mui/icons-material';
+import { AddRounded } from '@mui/icons-material';
 import { UserStatus } from '@communecar/types';
 import { useCallback } from 'react';
 import { statusIcons } from '../../utils/communities/userStatusIcons';
 
 export interface CommunityCardProps {
-  setJoined: React.Dispatch<React.SetStateAction<boolean>>;
+  onJoinRequest: () => void;
```

Comment:
If this is a generic status button calling this function `onJoinRequest` when it handles leaving the community too is a bit weird

Reply:
Let's say you changed jobs or left a place where you worked